pkg/iacinventory/github.go

```diff
@@ -0,0 +1,241 @@
+		}
+		var cfdirs []string
+		for dir := range repo.CloudformationDirs {
+			cfdirs = append(cfdirs, dir)
+		}
+		var cis []string
+		for _, ci := range repo.CI {
+			cis = append(cis, string(ci))
+		}
+		out = append(out, Repo{
+			Name:               name,
+			CIs:                cis,
+			TerraformDirs:      tfdirs,
+			CloudformationDirs: cfdirs,
+		})
+	}
+
+	pretty, err := json.MarshalIndent(out, "", "    ")
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal JSON: %w", err)
+	}
+
+	// print some nice, non-json info about notable repos
+	for _, repo := range out {
+		ciMsg := "and has NO CI configuration."
+		if len(repo.CIs) == 1 {
+			ciMsg = fmt.Sprintf("and is configured with %s CI.", repo.CIs[0]) // TODO: include other CI systems
+		}
+		if len(repo.CIs) > 1 {
+			ciMsg = "and is configured with multiple CI systems."
+		}
+		if len(repo.TerraformDirs) > 0 {
+			log.Infof("[%s]: contains %d Terraform directories %s\n", repo.Name, len(repo.TerraformDirs), ciMsg)
+		}
+		if len(repo.CloudformationDirs) > 0 {
+			log.Infof("[%s]: contains %d CloudFormation directories %s\n", repo.Name, len(repo.CloudformationDirs), ciMsg)
+		}
+	}
+
+	return jnode.FromJSON(pretty)
+}
+
+func (g *GithubScanner) Stop() error {
+	return nil
+}
+
+// githubCredsFromFS reads the `gh` configuration file on disk to get GitHub credentials.
+func githubCredsFromFS() (username string, oauthToken string, retErr error) {
+	var ghConfig map[string]interface{}
+	var ghConfigFile string
+
+	user, err := user.Current()
+	if err != nil {
+		retErr = fmt.Errorf("unable to get OS user: %w", err)
+		return
+	}
+
+	// Github credentials can exist in one of two files: ~/.config/gh/hosts.yml,
+	// or ~/.config/gh/config.yml. Creds _should_ only exist in the latter, but
+	// on some systems it appears in the former for whatever reason.
+	ghConfigFile = filepath.Join(user.HomeDir, ".config/gh/hosts.yml")
+	if _, err := os.Stat(filepath.Join(user.HomeDir, ".config/gh/hosts.yml")); err != nil {
+		if !os.IsNotExist(err) {
+			retErr = fmt.Errorf("unable to get gh config: %w", err)
+			return
+		}
+		// file does not exist, try another
+		ghConfigFile = filepath.Join(user.HomeDir, ".config/gh/config.yml")
+		if _, err := os.Stat(filepath.Join(user.HomeDir, ".config/gh/config.yml")); err != nil {
+			if !os.IsNotExist(err) {
+				retErr = fmt.Errorf("unable to get gh config: %w", err)
+				return
+			}
+			// still no file
+			retErr = fmt.Errorf("unable to locate GitHub `gh` configuration file")
+			return
+		}
+	}
+	yamlf, err := ioutil.ReadFile(ghConfigFile)
+	if err != nil {
+		retErr = fmt.Errorf("unable to read gh config: %w", err)
+		return
+	}
+	if err := yaml.Unmarshal(yamlf, &ghConfig); err != nil {
+		retErr = fmt.Errorf("unable to parse gh config: %w", err)
+		return
+	}
+
+	// Depending on the "type" of gh configuration (see file selector above),
+	// the yaml will be structured differently
+	if conf := ghConfig["github.com"]; conf != nil {
```

Comment:
You might prefer using jnode for this: n := jnode.FromMap(ghConfig)

Then n.Path("github.com").Path("user").GetString() etc